### PR TITLE
Snd 151 create helm chars to deploy open imis bb to the sandbox

### PR DESCRIPTION
**Pull request by oleksiidn**

### CODE REVIEW

File: sandbox-open-imis/charts/backend/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v2
+name: backend
+description: A Helm chart for deploying a pre-configured X-Road Central Server
```
Comment (bertvi) [inline sandbox-open-imis/charts/backend/Chart.yaml]: Probably you are not deploying X-Road Central Server, is it a copy-paste error?

**Comment by bertvi** [comment]
I would suggest that you would consider using values and templates of Helm charts, so that values to be used in the templates are provided in the values files and templates contain variables - this way it would be easier to make changes, you would only need to change the "values.yaml" file instead of making changes to all the templates.

**Comment by oleksiidn** [comment]
> I would suggest that you would consider using values and templates of Helm charts, so that values to be used in the templates are provided in the values files and templates contain variables - this way it would be easier to make changes, you would only need to change the "values.yaml" file instead of making changes to all the templates.

opneIMIS has same deployments for backend and gateway. I used variables to reduce duplicates.
 https://github.com/GovStackWorkingGroup/sandbox-open-imis/pull/1/commits/9b02b62a56ba3782baa2690cfe4168864898d0c2

File: sandbox-open-imis/charts/gateway/values.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+deployment:
+  name: open-imis-backned
```
Comment (bertvi) [inline sandbox-open-imis/charts/gateway/values.yaml]: Shouldn't this be "open-imis-gateway"?

File: sandbox-open-imis/charts/gateway/templates/service.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: open-imis-gateway
```
Comment (bertvi) [inline sandbox-open-imis/charts/gateway/templates/service.yaml]: Why not use variables from values file here as well like for deployment?

File: sandbox-open-imis/charts/db/templates/secret.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: open-imis-secret
```
Comment (bertvi) [inline sandbox-open-imis/charts/db/templates/secret.yaml]: Why not use variables from values file here as well like from deployment?

File: sandbox-open-imis/charts/db/templates/db.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+apiVersion: rds.services.k8s.aws/v1alpha1
+kind: DBInstance
+metadata:
+  name: open-imis-db
```
Comment (bertvi) [inline sandbox-open-imis/charts/db/templates/db.yaml]: Why not use variables from values file here as well like from deployment?

File: sandbox-open-imis/charts/backend/values.yaml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+deployment:
+  name: open-imis-gateway
```
Comment (bertvi) [inline sandbox-open-imis/charts/backend/values.yaml]: Should probably be "open-imis-backend"

**Review by bertvi — CHANGES_REQUESTED** [review]

File: sandbox-open-imis/charts/backend/templates/deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Values.deployment.name }}
+  namespace: {{ .Values.deployment.namespace }}
```
Comment (laurivosandi) [inline sandbox-open-imis/charts/backend/templates/deployment.yaml]: Helm standard is to use {{ .Release.Namespace }} see https://helm.sh/docs/chart_template_guide/builtin_objects/

File: sandbox-open-imis/charts/backend/templates/deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+      run: {{ .Values.deployment.spec.name }}
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        run: {{ .Values.deployment.spec.name }}
+    spec:
+      containers:
+        - name: {{ .Values.deployment.containers.name }}
+          image: {{ .Values.deployment.containers.image }}
```
Comment (laurivosandi) [inline sandbox-open-imis/charts/backend/templates/deployment.yaml:18]: I think convention is to use {{ .Values.image }} directly unless you have multiple images in same deployment

File: sandbox-open-imis/charts/backend/templates/deployment.yaml (hunk 7 above)
Comment (oleksiidn) [reply sandbox-open-imis/charts/backend/templates/deployment.yaml]: Good point. Thank you for a feedback.
We have other similar charts so before applying best practice need to make a team opinion.
https://govstack-global.atlassian.net/browse/SND-199

File: sandbox-open-imis/charts/backend/templates/deployment.yaml (hunk 8 above)
Comment (oleksiidn) [reply sandbox-open-imis/charts/backend/templates/deployment.yaml:18]: Good point too. 
But before applying it need to have team opinion regards other similar cases.
https://govstack-global.atlassian.net/browse/SND-205
Thank you for a feedback.

File: sandbox-open-imis/charts/backend/Chart.yaml (hunk 1 above)
Comment (oleksiidn) [reply sandbox-open-imis/charts/backend/Chart.yaml]: Yep, copy-paste error. Thank you.

File: sandbox-open-imis/charts/gateway/values.yaml (hunk 2 above)
Comment (oleksiidn) [reply sandbox-open-imis/charts/gateway/values.yaml]: `open-imis-gateway` correct  :+1:

File: sandbox-open-imis/charts/backend/values.yaml (hunk 6 above)
Comment (oleksiidn) [reply sandbox-open-imis/charts/backend/values.yaml]: `open-imis-backend` correct  :+1:

File: sandbox-open-imis/charts/gateway/templates/service.yaml (hunk 3 above)
Comment (oleksiidn) [reply sandbox-open-imis/charts/gateway/templates/service.yaml]: I am new to Helm Charts and it's not obvious for me when it is better to use variables.
I applied suggestion.
Thank you. 
https://github.com/GovStackWorkingGroup/sandbox-open-imis/pull/1/commits/59675dc95eb546ced4c6089484c6a1bebd8f7fb1

File: sandbox-open-imis/charts/db/templates/secret.yaml (hunk 4 above)
Comment (oleksiidn) [reply sandbox-open-imis/charts/db/templates/secret.yaml]: Applied suggestion.

File: sandbox-open-imis/charts/db/templates/db.yaml (hunk 5 above)
Comment (oleksiidn) [reply sandbox-open-imis/charts/db/templates/db.yaml]: Applied suggestion.

**Review by bertvi — APPROVED** [review]

**Pull request merged**